Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt

```diff
@@ -56,20 +64,34 @@ class SurveyFragment : Fragment() {
             )
             setContent {
                 JetsurveyTheme {
-                    viewModel.uiState.observeAsState().value?.let { surveyState ->
-                        when (surveyState) {
+                    val state = viewModel.uiState.observeAsState().value ?: return@JetsurveyTheme
+                    AnimatedContent(
+                        targetState = state,
+                        transitionSpec = {
+                            fadeIn() + slideIntoContainer(
+                                towards = AnimatedContentScope
+                                    .SlideDirection.Up,
+                                animationSpec = tween(600)
+                            ) with
+                                fadeOut(animationSpec = tween(200))
```

Comment:
Minor nit that the 600 on line 74 and 200 on line 76 feel like magic numbers. It'd be cool to have them defined in the theme and grab them as composition locals, but maybe it's enough to just make them constants since they're only used here. I'm also wondering if these two numbers have a relationship, so it can be something like fastAnimationDuration = 200, slowAnimationDuration = fastAnimationDuration * 3? Or if they should be independent.

Reply:
They should be independent - i've added two constants.